instrumentation/armeria-1.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/armeria/v1_3/ArmeriaSingletons.java

```diff
@@ -35,7 +35,12 @@ public final class ArmeriaSingletons {
             .build();
 
     CLIENT_DECORATOR = telemetry.newClientDecorator();
-    SERVER_DECORATOR = telemetry.newServiceDecorator();
+    SERVER_DECORATOR = wrapResponseCustomizer(telemetry.newServiceDecorator());
+  }
+
+  private static Function<? super HttpService, ? extends HttpService> wrapResponseCustomizer(
+      Function<? super HttpService, ? extends HttpService> function) {
+    return function.compose(service -> new ResponseCustomizerDecorator(service));
```

Comment:
Can you do this, or is there some type trick happening?

```suggestion
    SERVER_DECORATOR = telemetry.newServiceDecorator().compose(s -> new ResponseCustomizerDecorator(s));
  }
```

Reply:
Thanks, changed it